static/app/views/settings/organizationDeveloperSettings/sentryApplicationDashboard/requestLog.tsx

```diff
@@ -110,187 +110,157 @@ function TimestampLink({date, link}: {date: moment.MomentInput; link?: string})
   );
 }
 
-type Props = DeprecatedAsyncComponent['props'] & {
+interface RequestLogProps {
   app: SentryApp;
-};
+}
 
-type State = DeprecatedAsyncComponent['state'] & {
-  currentPage: number;
-  errorsOnly: boolean;
-  eventType: string;
-  requests: SentryAppWebhookRequest[];
-};
+function makeRequestLogQueryKey(slug: string): ApiQueryKey {
+  return [`/sentry-apps/${slug}/requests/`];
+}
 
-export default class RequestLog extends DeprecatedAsyncComponent<Props, State> {
-  shouldReload = true;
+export default function RequestLog({app}: RequestLogProps) {
+  const [currentPage, setCurrentPage] = useState<number>(0);
+  const [errorsOnly, setErrorsOnly] = useState<boolean>(false);
```

Comment:
TypeScript can usually infer the generic from the default value. You might not need to manually type these?